pam_fprint_grosshack: init at 0.3.0

agvantibo-again opened the pull request:
pam_fprint_grosshack: pam fprint module fork for subjectively better UX for fprint authentication;
fixes #395807
Upstream and readme: https://gitlab.com/mishakmak/pam-fprint-grosshack

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html)) _# out of scope_
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,61 @@
+  version = "0.3.0";
+  outputs = [ "out" ];
+
+  src = fetchurl {
+    url = "https://gitlab.com/mishakmak/pam-fprint-grosshack/-/archive/v${version}/pam-fprint-grosshack-v0.3.0.tar.gz";
+    sha256 = "sha256-5NiVV0o0dkc6SsWgKT09/zgMw8lZtWTlSUqV2qnYsV8=";
+  };
```
why can't we use fetch from gitlab?
```suggestion
  src = fetchFromGitLab {
    owner = "mishakmak";
    repo = "pam-fprint-grosshack";
    tag = "v${version}";
    sha256 = "sha256-5NiVV0o0dkc6SsWgKT09/zgMw8lZtWTlSUqV2qnYsV8=";
  };
```

agvantibo-again replied:
That's a great idea, will implement it. Didn't really know about fetchFromGitlab to be honest.

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,61 @@
+    "-Dsystemd_system_unit_dir=${placeholder "out"}/lib/systemd/system"
+  ];
+
+  PKG_CONFIG_DBUS_1_INTERFACES_DIR = "${placeholder "out"}/share/dbus-1/interfaces";
+  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
+  PKG_CONFIG_DBUS_1_DATADIR = "${placeholder "out"}/share";
```
```suggestion
  env = {
    PKG_CONFIG_DBUS_1_INTERFACES_DIR = "${placeholder "out"}/share/dbus-1/interfaces";
    PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
    PKG_CONFIG_DBUS_1_DATADIR = "${placeholder "out"}/share";
  };
```

0xda157 reviewed (commented):
also if you put `fixes #<issue-number>` in the description of this pr github will auto resolve the issue when this pr is merged

agvantibo-again commented:
@awwpotato thank you for the input, I implemented the changes you proposed. Would it be wise to squash the commits, or is the PR fine as-is?

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,65 @@
+  dbus,
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "pam_fprint_grosshack";
+  version = "0.3.0";
+  outputs = [ "out" ];
+
+  src = fetchFromGitLab {
+    owner = "mishakmak";
+    repo = "pam-fprint-grosshack";
+    tag = "v${version}";
```
finalAttrs is preferred in this case 
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "pam_fprint_grosshack";
  version = "0.3.0";
  outputs = [ "out" ];

  src = fetchFromGitLab {
    owner = "mishakmak";
    repo = "pam-fprint-grosshack";
    tag = "v${finalAttrs.version}";
```

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,65 @@
+
+  pname = "pam_fprint_grosshack";
+  version = "0.3.0";
+  outputs = [ "out" ];
```
since this derivation only has one output we don't need the `outputs` setting
```suggestion
```

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,65 @@
+  meta = {
+    description = "PAM Fprint module which implements the simultaneous password and fingerprint behaviour";
+    homepage = "https://gitlab.com/mishakmak/pam-fprint-grosshack";
+    license = lib.licenses.gpl2Only;
```
this should be plus based on this part at the top of each source file `either version 2 of the License, or (at your option) any later version`
```suggestion
    license = lib.licenses.gpl2Plus;
```

0xda157 reviewed (commented):
> Would it be wise to squash the commits, or is the PR fine as-is?

All the changes touching `pkgs/by-name/pa/pam_fprint_grosshack/` should be in one commit titled `pam_fprint_grosshack: init at 0.3.0` and adding yourself to the maintainer list should be in separate commit titled `maintainers: add agvantibo`.

agvantibo-again commented:
@awwpotato I applied the changes and formatted the file with nixfmt --strict, but ci seems to think otherwise:
https://github.com/NixOS/nixpkgs/actions/runs/14747590730/job/41397761393?pr=402506

The logs apparently refer to a different package. What should I do?

awwpotato commented:
> @awwpotato I applied the changes and formatted the file with nixfmt --strict, but ci seems to think otherwise: https://github.com/NixOS/nixpkgs/actions/runs/14747590730/job/41397761393?pr=402506
> 
> The logs apparently refer to a different package. What should I do?

seems like you figured it out, but the files should be formatted with `nixfmt-rfc-style`

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,64 @@
+    PKG_CONFIG_DBUS_1_DATADIR = "${placeholder "out"}/share";
+  };
+
+  meta = with lib; {
+    description = "PAM Fprint module which implements the simultaneous password and fingerprint behaviour";
+    homepage = "https://gitlab.com/mishakmak/pam-fprint-grosshack";
+    license = licenses.gpl2Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ agvantibo ];
```
with lib increase eval time and is best to avoid
```suggestion
  meta = with lib; {
    description = "PAM Fprint module which implements the simultaneous password and fingerprint behaviour";
    homepage = "https://gitlab.com/mishakmak/pam-fprint-grosshack";
    license = licenses.gpl2Plus;
    platforms = platforms.linux;
    maintainers = with maintainers; [ agvantibo ];
```
```suggestion
  meta = {
    description = "PAM Fprint module which implements the simultaneous password and fingerprint behaviour";
    homepage = "https://gitlab.com/mishakmak/pam-fprint-grosshack";
    license = lib.licenses.gpl2Plus;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ agvantibo ];
```

agvantibo-again committed:
pam_fprint_grosshack: init at 0.3.0

pam_fprint_grosshack: pam fprint module fork
with subjectively better UX for fprint authentication;
fixes #395807
Upstream: https://gitlab.com/mishakmak/pam-fprint-grosshack

awwpotato commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 402506`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pam_fprint_grosshack</li>
  </ul>
</details>

awwpotato commented on pkgs/by-name/pa/pam_fprint_grosshack/package.nix:
```diff
@@ -0,0 +1,64 @@
+    PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
+    PKG_CONFIG_DBUS_1_DATADIR = "${placeholder "out"}/share";
+  };
+
```
can be tested with `nix-update pam_fprint_grosshack`
```suggestion

  passthru.updateScript = nix-update-script { };

```

agvantibo-again replied:
It appears nix-update chose to swap out the version for 0.1.0 instead of 0.3.0; how do I control this behavior?
`Update 0.3.0 -> 0.1.0 in /home/agvantibo/Downloads/Builds/nixpkgs/pkgs/by-name/pa/pam_fprint_grosshack/package.nix`

awwpotato replied:
nix-update doesn't work correctly on gitlab when the project doesn't have a release for each tag, we should leave out the update script for this package.
